src/khoj/main.py

```diff
@@ -118,6 +112,15 @@ def run():
 
         gui.exec()
 
+    if not state.demo:
```

Comment:
We only need to have a wake-up thread for scheduled jobs when khoj in "daemon" (i.e no gui) mode.
```suggestion
    if not args.gui and not state.demo:
```